src/main/kotlin/logic/usecase/GuessFoodPreparationTimeUseCase.kt

```
@@ -0,0 +1,52 @@
+package logic.usecase
+
+import model.GuessWrongPreparationTimeException
+import model.InValidPreparationTimeException
+import model.RichMaxAttemptException
+
+
+class GuessFoodPreparationTimeUseCase() {
+    private var attempts = 0
+    private var preparationTime: Int = 0
+
+
+    operator fun invoke(userGuess: Int): Result<String> {
+        attempts++
+        return if (userGuess == preparationTime)
+            onUserGuessCorrectTime()
+        else if (attempts == MAX_ATTEMPT)
+            onUserRichMaxAttempt(preparationTime)
+        else if (userGuess < preparationTime)
+            Result.failure(GuessWrongPreparationTimeException("The preparation time is too low."))
```

the same as described earlier, you can just throw an error instead of returning "Result" to UI